packages/nodes-base/utils/sendAndWait.utils.ts

```diff
@@ -0,0 +1,392 @@
+		},
+		...additionalProperties,
+		{
+			displayName:
+				'Use the wait node for more complex approval flows. <a href="https://docs.n8n.io/nodes/n8n-nodes-base.wait" target="_blank">More info</a>',
+			name: 'useWaitNotice',
+			type: 'notice',
+			default: '',
+		},
+	];
+
+	return updateDisplayOptions(
+		{
+			show: {
+				resource: [resource],
+				operation: [SEND_AND_WAIT_OPERATION],
+			},
+		},
+		sendAndWait,
+	);
+}
+
+// Webhook Function --------------------------------------------------------------
+export async function sendAndWaitWebhook(this: IWebhookFunctions) {
+	const query = this.getRequestObject().query as { approved: 'false' | 'true' };
+	const approved = query.approved === 'true';
+	return {
+		webhookResponse: ACTION_RECORDED_PAGE,
+		workflowData: [[{ json: { data: { approved } } }]],
+	};
+}
+
+// Send and Wait Config -----------------------------------------------------------
+export function getSendAndWaitConfig(context: IExecuteFunctions): SendAndWaitConfig {
+	const message = escapeHtml((context.getNodeParameter('message', 0, '') as string).trim());
+	const subject = escapeHtml(context.getNodeParameter('subject', 0, '') as string);
+	const resumeUrl = context.evaluateExpression('{{ $execution?.resumeUrl }}', 0) as string;
+	const nodeId = context.evaluateExpression('{{ $nodeId }}', 0) as string;
+	const approvalOptions = context.getNodeParameter('approvalOptions.values', 0, {}) as {
+		approvalType?: 'single' | 'double';
+		approveLabel?: string;
+		buttonApprovalStyle?: string;
+		disapproveLabel?: string;
+		buttonDisapprovalStyle?: string;
+	};
+
+	const config: SendAndWaitConfig = {
+		title: subject,
+		message,
+		url: `${resumeUrl}/${nodeId}`,
+		options: [],
+	};
+
+	if (approvalOptions.approvalType === 'double') {
+		const approveLabel = escapeHtml(approvalOptions.approveLabel || 'Approve');
+		const buttonApprovalStyle = approvalOptions.buttonApprovalStyle || 'primary';
+		const disapproveLabel = escapeHtml(approvalOptions.disapproveLabel || 'Disapprove');
+		const buttonDisapprovalStyle = approvalOptions.buttonDisapprovalStyle || 'secondary';
+
+		config.options.push({
+			label: disapproveLabel,
+			value: 'false',
+			style: buttonDisapprovalStyle,
+		});
+		config.options.push({
+			label: approveLabel,
+			value: 'true',
+			style: buttonApprovalStyle,
+		});
+	} else {
+		const label = escapeHtml(approvalOptions.approveLabel || 'Approve');
+		const style = approvalOptions.buttonApprovalStyle || 'primary';
+		config.options.push({
+			label,
+			value: 'true',
+			style,
+		});
+	}
+
+	return config;
+}
+
+// Create Email Helpers ----------------------------------------------------------------------------------
+
+function createEmailBody(message: string, buttons: string, instanceId?: string) {
+	const utm_campaign = instanceId ? `&utm_campaign=${instanceId}` : '';
+	const n8nWebsiteLink = `https://n8n.io/?utm_source=n8n-internal&utm_medium=send-and-wait${utm_campaign}`;
+	return `
+<!DOCTYPE html>
```

Comment:
having html code inline makes this code very hard to read, can you extract this to a different file?